library/src/main/java/com/novoda/downloadmanager/DownloadFile.java

```diff
@@ -114,26 +115,20 @@ public void onDownloadFinished() {
         });
     }
 
-    private Error convertError(FilePersistenceResult status) {
+    private DownloadError convertError(FilePersistenceResult status) {
         switch (status) {
-            case SUCCESS:
-                Logger.e("Cannot convert success status to any DownloadError type");
-                break;
             case ERROR_UNKNOWN_TOTAL_FILE_SIZE:
-                return DownloadError.Error.FILE_TOTAL_SIZE_REQUEST_FAILED;
+                return DownloadErrorFactory.createTotalSizeRequestFailedError(downloadFileId, url);
             case ERROR_INSUFFICIENT_SPACE:
-                return DownloadError.Error.FILE_CANNOT_BE_CREATED_LOCALLY_INSUFFICIENT_FREE_SPACE;
+                return DownloadErrorFactory.createInsufficientFreeSpaceError(downloadFileStatus);
             case ERROR_EXTERNAL_STORAGE_NON_WRITABLE:
-                return DownloadError.Error.STORAGE_UNAVAILABLE;
+                return DownloadErrorFactory.createStorageNotAvailableError(filePersistence);
             case ERROR_OPENING_FILE:
-                return DownloadError.Error.FILE_CANNOT_BE_WRITTEN;
+                return DownloadErrorFactory.createCannotWriteToFileError(downloadFileStatus);
             default:
                 Logger.e("Status " + status + " missing to be processed");
-                break;
-
+                return DownloadErrorFactory.createUnknownError();
```

Comment:
I'd add the unknown status here

Reply:
I don't know what that means 😂 the `Type` is added in the factory.